pkg/cli/clusteroperator/waitforstable/command.go

```diff
@@ -0,0 +1,148 @@
+	return &WaitForStableOptions{
+		RESTClientGetter:    restClientGetter,
+		Timeout:             1 * time.Hour,
+		MinimumStablePeriod: 5 * time.Minute,
+		waitInterval:        10 * time.Second,
+
+		IOStreams: streams,
+	}
+}
+
+// NewCmdLogout implements a command to wait for OpenShift platform operators to become stable
+func NewCmdWaitForStableClusterOperators(restClientGetter genericclioptions.RESTClientGetter, streams genericclioptions.IOStreams) *cobra.Command {
+	o := NewWaitForStableOptions(restClientGetter, streams)
+
+	cmd := &cobra.Command{
+		Use:     "wait-for-stable-cluster",
+		Short:   "wait for the platform operators to become stable",
+		Long:    waitForStableLong,
+		Example: waitForStableExample,
+		Run: func(cmd *cobra.Command, args []string) {
+			kcmdutil.CheckErr(o.Complete())
+			kcmdutil.CheckErr(o.Run(context.Background()))
+		},
+	}
+
+	o.AddFlags(cmd)
+
+	return cmd
+}
+
+func (o *WaitForStableOptions) AddFlags(cmd *cobra.Command) error {
+	cmd.Flags().DurationVar(&o.Timeout, "timeout", o.Timeout, "duration before the command times out. Defaults to 1 hour.")
+	cmd.Flags().DurationVar(&o.MinimumStablePeriod, "minumum-stable-period", o.MinimumStablePeriod, "minimum duration to consider a cluster stable. Defaults to 5 minutes.")
+
+	return nil
+}
+
+func (o *WaitForStableOptions) Complete() error {
+	if o.Timeout <= o.MinimumStablePeriod {
+		return fmt.Errorf("--timeout must be greater than the --minimum-stable-period")
+	}
+
+	clientConfig, err := o.RESTClientGetter.ToRESTConfig()
+	if err != nil {
+		return err
+	}
+
+	configClient, err := configv1client.NewForConfig(clientConfig)
+	if err != nil {
+		return err
+	}
+	o.configClient = configClient
+
+	return nil
+}
+
+func (o WaitForStableOptions) Run(ctx context.Context) error {
+
+	if o.Timeout > 0 {
+		var cancel func()
+		ctx, cancel = context.WithTimeout(ctx, o.Timeout)
+		defer cancel()
+	}
+
+	var stableFor time.Duration
+	waitErr := wait.PollImmediateUntilWithContext(ctx, o.waitInterval, func(waitCtx context.Context) (done bool, err error) {
+		defer func() {
+			if !done {
+				stableFor = 0
+			}
+			if stableFor < o.MinimumStablePeriod {
+				done = false
+			}
+		}()
+		stableFor += o.waitInterval
+
+		operators, err := o.configClient.ClusterOperators().List(waitCtx, metav1.ListOptions{})
+		if err != nil {
+			fmt.Fprintf(o.ErrOut, "failed to list clusteroperators: %v", err)
+			return false, nil
+		}
+
+		for _, operator := range operators.Items {
+			if unstableReason := unstableOperatorReason(&operator); len(unstableReason) > 0 {
+				fmt.Fprintf(o.Out, "operator %q is not yet stable: %v\n", operator.Name, unstableReason)
+				return false, nil
+			}
+		}
+
+		return true, nil
+	})
+
+	return waitErr
+}
+
+func unstableOperatorReason(operator *configv1.ClusterOperator) string {
+	if !v1helpers.IsStatusConditionTrue(operator.Status.Conditions, configv1.OperatorAvailable) {
+		return "available != true"
```

Comment:
we may twiddle these in the future, but the conditions look correct now.

Reply:
I had to read this twice since we're returning something that *looks like (Go) code* but is actually for humans?  Why not something more like
`fmt.Sprintf("operator %s is not available", operator.name)` etc?